Lib/string.py

```diff
@@ -157,6 +157,10 @@ def convert(mo):
         return self.pattern.sub(convert, self.template)
 
 
+# We use re.I | re.A when compiling Template.idpattern, but restore old flag
+# for backward compatibility.
+Template.flags = _re.IGNORECASE
```

Comment:
How about:

> We use re.I | re.A while compiling Template.idpattern in the metaclass above, but since
> flags is part of the public API, we restore its original documented value for backward
> compatibility.

?